client/components/Main/SearchPanel.tsx

```diff
@@ -23,6 +23,7 @@ interface IProps {
 
 interface IState {
     params: ISearchParams;
+    key?: string;
```

Comment:
`key` should be a required field as it will always have a value. Therefor the `?` suffix isn't required